Source/Factories/JSONFactory.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  JSONFactory.swift
+//  Matrioska
+//
+//  Created by Mathias Aichinger on 11/01/2017.
+//  Copyright © 2017 runtastic. All rights reserved.
+//
+
+import Foundation
+
+/// An error type for JSONFactory
+///
+/// - missing: specifies that a mandatory key is missing
+enum JSONFactoryError: Error {
+    case missing(JSONObject, String)
+}
+
+/// A factory that wraps ComponentFactory objects and uses them to produce Components
+public final class JSONFactory {
+    
+    static let typeKey = "type"
+    static let metaKey = "meta"
+    static let childrenKey = "children"
```

Comment:
For now we agreed on having the same `children` key for both wrapper and cluster afaik